src/main/java/org/zalando/nakadi/plugin/api/authz/AuthorizationService.java

```diff
@@ -0,0 +1,33 @@
+package org.zalando.nakadi.plugin.api.authz;
+
+import org.zalando.nakadi.plugin.api.PluginException;
+
+public interface AuthorizationService {
+
+    enum Operation {
+        READ, WRITE, ADMIN
+    }
+
+    /**
+     * Check whether a caller, represented by a token, is authorized to perform an operation on a resource.
+     *
+     * @param subject The subject that performs the operation on the resource
+     * @param operation the operation (read, write, admin) to authorize
+     * @param resource the resource that the subject wants to perform an operation on
+     * @return true if the subject, is authorized to perform the operation on the resource
+     * @throws PluginException if an error occurred during execution
+     */
+    boolean isAuthorized(Subject subject, Operation operation, Resource resource);
```

Comment:
Is there any particular reason why you have not defined exception in the signature ?